bpf/init.sh

```diff
@@ -481,10 +482,10 @@ else
 	ip link del cilium_sit   2> /dev/null || true
 fi
 
-if [ "$MODE" = "vxlan" -o "$MODE" = "geneve" ]; then
-	ENCAP_DEV="cilium_${MODE}"
+if [ "$MODE" = "tunnel" ]; then
```

Comment:
Could you double-check the rest of the `$MODE` usage in the file? It looks like there are places where it could be `ipip`, which I presume should be handled via the `${TUNNEL_MODE}` option as well.

Reply:
Yes. `$MODE` could be `ipip`. It's not used for tunnel mode, but it should be covered. @MasterZ40 can you check it out and fix it?

Reply:
Thanks for the comments. 
Sorry if i did not fully understood, but it seems like `ipip` mode is handled separately in this [file.](https://github.com/cilium/cilium/blob/72949b4b49d43e6dbb4714929222acb688ce6dd2/bpf/init.sh#L441) @borkmann could you comment?

Reply:
@MasterZ40 The condition here should check for VXLAN or GENEVE. We don't want to create a `cilium_ipip` device for IPIP.

Reply:
@MasterZ40 the ipip is handled separately, mostly given we need separate devices for v4/v6 which is not the case for vxlan + geneve. Also the kernel device name handling is more hacky for the ipip ones which ignores passed name when it needs to pull in the driver.

For testing ipip you can use: `./daemon/cilium-agent --enable-ipv4=true --enable-ipv6=true --datapath-mode=lb-only --bpf-lb-algorithm=maglev --bpf-lb-maglev-table-size=2039 --bpf-lb-mode=dsr --bpf-lb-acceleration=native --devices=enp2s0np0 --bpf-lb-dsr-dispatch=ipip --disable-envoy-version-check=true`

Reply:
@borkmann thanks for the clarification.  @pchaigno I think in this case as Daniel said the ipip devices are handled separately as datapath tunnel devices. And this change is not really touching that separation. The ipip devices are handled here: https://github.com/cilium/cilium/blob/72949b4b49d43e6dbb4714929222acb688ce6dd2/bpf/init.sh#L441 

Let me know if this makes sense. Thx!

Reply:
Ah, I thought IPIP would be one of the tunnel modes, but it's actually one of the modes. Ok :+1: